lib/db2.js

```diff
@@ -41,7 +42,7 @@ function DB2(settings) {
   this.supportColumnStore = (settings.supportColumnStore || false);
   this.supportDashDB = (settings.supportDashDB || false);
 
-  this.connStr =
+  this.connStrGenerate =
```

Comment:
@jannyHou is there any particular reason why to expose the generated connection string and also `this.dsn`? I'd personally just use local variables:

```js
var dsn = settings.dsn;
var connStrGenerate = // ...
this.connStr = dsn || connStrGenerate;
```

One could argue that it's not necessary to build the connection string when `dsn` is provided, i.e.

```js
if (settings.dsn) {
  this.connStr = settings.dsn;
} else {
  this.connStr = // build the string
}
```

Thoughts?